pkg/controller/replication_controller.go

```diff
@@ -351,16 +351,20 @@ func (rm *ReplicationManager) syncReplicationController(key string) error {
 	}
 	controller := *obj.(*api.ReplicationController)
 
+	// Check the expectations of the rc before counting active pods, otherwise a new pod can sneak in
+	// and update the expectations after we've retrieved active pods from the store. If a new pod enters
+	// the store after we've checked the expectation, the rc sync is just deferred till the next relist.
```

Comment:
OK-- sorry for the delay reviewing, it took me some time to think through this. I think you're right.